### PR TITLE
RGB Matrix support for TBK Mini

### PR DESCRIPTION
**Pull request by filterpaper**
## Description

Add RGB matrix support for BastardKB's TBK Mini

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [x] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by filterpaper** [comment]
@bstiq Please do review and feedback if there are any corrections.

File: keyboards/bastardkb/tbkmini/rules.mk [hunk 1]
```diff
@@ -17,7 +17,10 @@ SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
 # if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
 NKRO_ENABLE = yes            # USB Nkey Rollover
 BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
-RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+RGB_MATRIX_ENABLE = yes     # Enable keyboard RGB matrix (do not use together with RGBLIGHT_ENABLE)
```
Comment (drashna) [inline keyboards/bastardkb/tbkmini/rules.mk:22]: `RGB_MATRIX_SUPPORTED` needs to be set to "yes", as well.

Comment (filterpaper) [reply keyboards/bastardkb/tbkmini/rules.mk:22]: Thanks for the catch.

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/bastardkb/tbkmini/rules.mk (hunk 1 above)
Comment (drashna) [reply keyboards/bastardkb/tbkmini/rules.mk:22]: Welcome!  Otherwise, the "no" for the supported will forcibly deactivate the rgb matrix feature. :D

**Comment by drashna** [comment]
@bstiq Any issues with this?  If not, would like to go ahead and merge it

**Comment by bstiq** [comment]
@drashna thanks for the checks.
I haven't had time to check it properly yet. This week I will build a skeletyl with RGB and check it works.
Then I will comment back on this issue to confirm.

If someone else has a skeletyl and can confirm, that also works for me.

Regards,

**Comment by bstiq** [comment]
@drashna  Tested and confirmed, you can merge.
Thank you for the swift review, and @filterpaper for your work.

Regards,

**Comment by drashna** [comment]
Thanks!

**Pull request merged**